### PR TITLE
 Add @SSZSerializable.serializeAs annotation specifier 

### PR DESCRIPTION
**Pull request by Nashatyrev**

### CODE REVIEW

File: ssz/src/main/java/org/ethereum/beacon/ssz/SSZSchemeBuilder.java [hunk 1]
```diff
@@ -36,6 +37,21 @@
        */
       public boolean notAContainer = false;
 
+      /**
+       *  If the field class specifies {@link SSZSerializable#serializeAs()} attribute
+       *  returns the specified class.
+       *  Else returns type value.
+       */
+      public Class<?> getSerializableClass() {
```
Comment (zilm13) [inline ssz/src/main/java/org/ethereum/beacon/ssz/SSZSchemeBuilder.java]: I'd put this logic somewhere in `SSZSchemeBuilder`
The idea behind these classes was that `SSZField` just describes the scheme of object

File: ssz/src/main/java/org/ethereum/beacon/ssz/type/SubclassCodec.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import net.consensys.cava.ssz.BytesSSZReaderProxy;
+import org.ethereum.beacon.ssz.ConstructorObjCreator;
+import org.ethereum.beacon.ssz.SSZSchemeBuilder.SSZScheme.SSZField;
+
+public class SubclassCodec implements SSZCodec {
```
Comment (zilm13) [inline ssz/src/main/java/org/ethereum/beacon/ssz/type/SubclassCodec.java]: A bit of Javadoc here, please

File: ssz/src/main/java/org/ethereum/beacon/ssz/type/SubclassCodec.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      SSZField field, OutputStream result) {
+    superclassCodec.encodeList(value, getSerializableField(field), result);
+  }
+
+  @Override
+  public Object decode(SSZField field,
+      BytesSSZReaderProxy reader) {
+    SSZField serializableField = getSerializableField(field);
+    Object serializableTypeObject = superclassCodec.decode(serializableField, reader);
+    return ConstructorObjCreator.createInstanceWithConstructor(
```
Comment (zilm13) [inline ssz/src/main/java/org/ethereum/beacon/ssz/type/SubclassCodec.java]: Ouch
I have not seen this!

Comment (Nashatyrev) [reply ssz/src/main/java/org/ethereum/beacon/ssz/type/SubclassCodec.java]: What do you mean?

**Pull request merged**